README.md

```diff
@@ -1,37 +1,56 @@
+2. [Getting started with nyaml](#how-to-use-nyaml-tool)
+3. [Workflow](#nyaml-workflow)
+4. [How to use nyaml](#how-to-use-nyaml-tool)
+5. [Conversion from YAML to XML](#conversion-from-yaml-to-xml)
+6. [Design of NeXusOntology and terms in YAML](#design-of-nexus-dataformat-and-terms-in-yaml)
+   - [NeXus root section for base classes and application definitions](#root-section-for-base-classes-and-application-definitions)
+   - [NeXus group](#nexus-group)
+   - [NeXus field and NeXus attribute](#nexus-field-and-nexus-attrubute)
+   - [NeXus link](#nexus-link)
+   - [NeXus choice](#nexus-choice)
+7. [Special keywords in YAML](#special-keywords-in-yaml)
    - [Keyword `exists`](#keyword-exists)
    - [Keyword `unit`](#keyword-unit)
    - [Keyword `dimensions`](#keyword-dimensions)
    - [Keyword `enumeration`](#keyword-enumeration)
    - [Keyword `xref`](#keywprd-xref)
-7. [How to Install nyaml](#how-to-use-nyaml-tool)
 8. [Conclusion](#conclusion)
 9. [References](#references)
 
-## Introduction
-
-The NeXus data format, described by the NeXus Definition Language (NXDL), represents a concerted effort aimed at facilitating data exchange within scientific communities, particularly among those engaged in neutron, X-ray, and muon research [J. Appl. Cryst. (2015). 48, 301-305](https://doi.org/10.1107/S1600576714027575). It serves as a standardized framework for both data exchange and storage. At its core, the NeXus Definition Language (NXDL) functions as the cornerstone through which scientists delineate the nomenclature and organizational structure of information within NeXus data files, tailored to specific scientific techniques.
 
-NXDL is used to define general data storage objects (base classes) and use them as the building blocks for defining measurement-specific or even instrument-specific data storage objects (application definitions). In this process, members and definitions of individual base classes can be used as is or customized. In essence, the process of schema development, whether for a base class or an application definition, entails crafting an NXDL schema definition file with the extension 'nxdl.xml', utilizing the Extensible Markup Language, [XML](https://www.w3.org/TR/REC-xml/REC-xml-20081126.xml) .
-
-To expedite the schema development process, we have introduced Yet Another Markup Language ([YAML](https://yaml.org/)), which provides a syntax or style specifically tailored for defining scientific domain-driven schemas with NXDL. One significant advantage of YAML over XML is its indentation-driven approach, which eliminates the need for starting and ending tags for each entity within the schema. The `YAML` format results in a reduction of NXDL keyword repetition and allows for a more intuitive grasp of Python syntax, such as class inheritance. These benefits are attained without compromising the integrity of the original NeXus schema, which is traditionally expressed in XML format.
+## Getting started with nyaml
```

Comment:
This should come close the beginning (maybe rename to installation) since we want people to use this tool.

Reply:
Agree on where to place the benefit, however, why not as bullet points rather than prose?